App/Sagas/ImageSharingTriggers.ts

```diff
@@ -0,0 +1,56 @@
+import { call, put, select } from 'redux-saga/effects'
+import { ActionType } from 'typesafe-actions'
+import RNFS from 'react-native-fs'
+
+import ProcessingImagesActions, { ProcessingImage, ProcessingImagesSelectors } from '../Redux/ProcessingImagesRedux'
+import UIActions from '../Redux/UIRedux'
+import { insertImage, addToIpfs, uploadArchive, shareWalletImage, addToWallet, shareToThread } from './ImageSharingSagas'
+
+export function * handleSharePhotoRequest(action: ActionType<typeof UIActions.sharePhotoRequest>) {
+  const { image, threadId, comment } = action.payload
+  if (!image || !threadId) {
+    return
+  }
+  if (typeof image === 'string') {
+    yield call(shareWalletImage, image, threadId, comment)
+  } else {
+    yield call(insertImage, image, threadId, comment)
+  }
+}
+
+export function * handleImageUploadComplete (action: ActionType<typeof ProcessingImagesActions.imageUploadComplete>) {
+  const { uuid } = action.payload
+  yield call(addToWallet, uuid)
+  try {
+    const processingImage: ProcessingImage | undefined = yield select(ProcessingImagesSelectors.processingImageByUuid, uuid)
+    if (processingImage && processingImage.addData && processingImage.addData.addResult.archive) {
+      const exists: boolean = yield call(RNFS.exists, processingImage.addData.addResult.archive.path)
+      if (exists) {
+        yield call(RNFS.unlink, processingImage.addData.addResult.archive.path)
+      }
+    }
+  } catch (e) {}
+}
+
+export function * retryImageShare (action: ActionType<typeof ProcessingImagesActions.retry>) {
```

Comment:
This one examines where in the process an error happened and retries from that point on.